Spec/Auth.swift

```diff
@@ -1106,6 +1106,36 @@ class Auth : QuickSpec {
                         done()
                     }
                 }
+                
+                tokenParams.clientId = "newClientId"
+                tokenParams.ttl = 2000
+                tokenParams.capability = "{ \"test:*\":[\"test\"] }"
+                
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.createTokenRequest(tokenParams, options: authOptions) { tokenRequest, error in
+                        expect(error).to(beNil())
+                        guard let tokenRequest = tokenRequest else {
+                            XCTFail("TokenDetails is nil"); done(); return
+                        }
+                        expect(tokenRequest.clientId).to(equal("newClientId"))
+                        expect(tokenRequest.ttl).to(equal(2000))
+                        expect(tokenRequest.capability).to(equal("{ \"test:*\":[\"test\"] }"))
+                        done()
+                    }
+                }
+                
+                tokenParams.clientId = nil
+                
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.createTokenRequest(tokenParams, options: authOptions) { tokenRequest, error in
```

Comment:
My concern with this commit is that at no point have you tested that `nil` AuthOptions does not change the AuthOptions, and at no point have you checked that `AuthOptions` overrides the defaults when provided. The assertions are all based on the token request properties.  You will need to test at least one attribute of AuthOptions please for completeness.